### ISSUE TITLE
Devise.config.send_email_changed_notification sending "Emailed Changed" email when user signs up

### ISSUE THREAD

**Issue by pmichaeljones** [plataformatec/devise#5047]
## Environment

- Ruby **2.4.5**
- Rails **4.2.11**
- Devise **4.6.1**

## Current behavior

Devise is sending the "Email Changed" email when a user first signs up. This email is unnecessary because they haven't confirmed this email yet.

GIF showing both "confirmation" email AND "email changed" email.

![Double Email Gif](https://user-images.githubusercontent.com/6728265/54702757-c90f0500-4b37-11e9-8e84-22fc9207b1ba.gif)

## Expected behavior

Devise should only send "Email Changed" email after user has signed up, confirmed and THEN changed their email.

**Comment by tegon** [comment]
Hi @pmichaeljones, thanks for reporting this.

I tried to reproduce the issue using a newly created application but I couldn't - I only received the confirmation email.
[email-changed-signup.zip](https://github.com/plataformatec/devise/files/3009890/email-changed-signup.zip)
I'm attaching the application I've used so that you can see if I'm missing something.

**Comment by pmichaeljones** [comment]
Hi @tegon. So I have an after_create hook in my user.rb model. It adds some data to a column after a user signs up, but before they confirm. I think this might be causing the issue. I'm going to explore the Devise gem and find where the confirmation email gets sent. I have a feeling it's after the first after_update call perhaps?

**Comment by pmichaeljones** [comment]
Here we go @tegon.

https://github.com/plataformatec/devise/blob/2e5b5fcd705b06c518ab0156b96badb91c4cb6ea/lib/devise/models/database_authenticatable.rb#L31

<img width="1057" alt="Screen Shot 2019-03-27 at 1 50 12 PM" src="https://user-images.githubusercontent.com/6728265/55077134-4c2fde00-5097-11e9-9388-490704853445.png">

Do you think it makes sense to do something like:

```
included do
    after_update :send_email_changed_notification, if: :send_email_changed_notification? && !user.confirmed
    [redacted]
end
```

In our app, we have an after_create hook that creates a specialized "code" for the user. This "updates" the user a millisecond after it was initial created. This "after_update" is then firing the "emailed_changed" email.

```
class User <  ActiveRecord::Base
    after_create :create_user_code
   
    private
   
    def create_user_code
        self.tracking_code = SecureRandom.hex(20)
    end
end
```

What do you think?

**Comment by pmichaeljones** [comment]
Any thoughts on this @tegon?

**Comment by tegon** [comment]
Sorry @pmichaeljones it's been really busy at work lately, I wasn't able to do much open source 😞 

I'll try to get back to you this week.

**Comment by pmichaeljones** [comment]
Tudo bom @tegon. Sem pressa. It's not that important of an issue. Just kind of a weird edge case that could potentially be fixed. I can offer up a PR if you'd like.

**Comment by tegon** [comment]
Yeah, I just wanted to understand exactly what's happening before thinking of a solution. It seems weird to me that the `email_changed?` method returns true since your code only updated the `tracking_code` column.

**Comment by ellery** [comment]
Sorry to drop in here on an old issue.

For anyone else that runs into this issue. Change to a before_create if possible to solve this problem. since the records object is still from the initial creation of the user email_changed will be true.

```
class User <  ActiveRecord::Base
    before_create :create_user_code
   
    private
   
    def create_user_code
        self.tracking_code = SecureRandom.hex(20)
    end
end
```

will allow you to add attributes without triggering the email_changed mailer.
